pkg/service/did/did.go

```diff
@@ -27,11 +28,40 @@ type Service struct {
 	keyStore *keystore.Service
 }
 
-// MethodHandler describes the functionality of *all* possible DID service, regardless of method
-type MethodHandler interface {
-	CreateDID(request CreateDIDRequest) (*CreateDIDResponse, error)
-	GetDID(request GetDIDRequest) (*GetDIDResponse, error)
-	GetDIDs(method didsdk.Method) (*GetDIDsResponse, error)
+func (s *Service) Type() framework.Type {
+	return framework.DID
+}
+
+// Status is a self-reporting status for the DID service.
+func (s *Service) Status() framework.Status {
```

Comment:
for these status stuff you could potentially loop through the service object fields using reflection and it could probably be less code, could almost make just one helper function
https://stackoverflow.com/questions/18926303/iterate-through-the-fields-of-a-struct-in-go

Reply:
you are correct, and this may be a good change. the only thing that would be missing is the relative error messages, but perhaps this is ok and we can surface the struct name.

let me open an issue for this. https://github.com/TBD54566975/ssi-service/issues/127